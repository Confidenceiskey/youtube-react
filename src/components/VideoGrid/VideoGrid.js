import React from 'react';
import './VideoGrid.scss';
import VideoGridHeader from './VideoGridHeader/VideoGridHeader';
import { Divider } from 'semantic-ui-react';
import VideoPreview from '../VideoPreview/VideoPreview';

const VideoGrid = ({ hideDivider, title }) => {
  const divider = hideDivider ? null : <Divider />;
  return (
    <React.Fragment>
      <VideoGridHeader title={title} />
      <div className='video-grid'>
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
      </div>
      {divider}
    </React.Fragment>
  );
}

export default VideoGrid;
