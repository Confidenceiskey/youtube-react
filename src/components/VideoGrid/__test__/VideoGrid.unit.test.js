import React from 'react';
import { shallow } from 'enzyme';
import VideoGrid from '../VideoGrid';

describe('VideoGrid', () => {
  it('renders VideoGrid w/o props', () => {
    const wrapper = shallow(
      <VideoGrid />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders VideoGrid w/ title prop', () => {
    const wrapper = shallow(
      <VideoGrid title='Trending' />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders VideoGrid w/o divider', () => {
    const wrapper = shallow(
      <VideoGrid hideDivider={true} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
