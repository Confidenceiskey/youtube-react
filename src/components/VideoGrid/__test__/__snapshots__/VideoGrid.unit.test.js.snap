// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoGrid renders VideoGrid w/ title prop 1`] = `
<Fragment>
  <VideoGridHeader
    title="Trending"
  />
  <div
    className="video-grid"
  >
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
  </div>
  <Divider />
</Fragment>
`;

exports[`VideoGrid renders VideoGrid w/o divider 1`] = `
<Fragment>
  <VideoGridHeader />
  <div
    className="video-grid"
  >
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
  </div>
</Fragment>
`;

exports[`VideoGrid renders VideoGrid w/o props 1`] = `
<Fragment>
  <VideoGridHeader />
  <div
    className="video-grid"
  >
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
  </div>
  <Divider />
</Fragment>
`;
